### PR TITLE
refactor(account): query-ify OrganizationContext TASK-1194

### PR DESCRIPTION
**Pull request by duvld**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [ ] Ensure the tests pass
4. [ ] Run `./python-format.sh` to make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/main/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes
8. [x] Create a testing plan for the reviewer and add it to the Testing section

## Description
Refactor some react context into react query.

## Notes

Basic refactoring example from our own implemented react context to react query. Using `OrganizatinContext` as the example.

## Testing

Go through functionality of billing/usage to see if the functionality remains the same. Look at code to see if react query is implemented correctly *and* if old react context has been removed correctly.

### CODE REVIEW

File: jsapp/js/constants.ts [hunk 1]
```diff
@@ -623,4 +623,10 @@ const constants = {
 
 export const HELP_ARTICLE_ANON_SUBMISSIONS_URL = 'managing_permissions.html';
 
+export enum ReactQueryStatuses {
```
Comment (jamesrkiger) [inline jsapp/js/constants.ts:626]: Is there a reason we can't just import `QueryStatus` from react-query wherever we need it?

File: jsapp/js/account/plans/plan.component.tsx [hunk 2]
```diff
@@ -115,8 +115,9 @@ export default function Plan(props: PlanProps) {
   >([]);
   const [products, loadProducts, productsStatus] = useContext(ProductsContext);
   useRefreshApiFetcher(loadProducts, productsStatus);
-  const [organization, loadOrg, orgStatus] = useContext(OrganizationContext);
-  useRefreshApiFetcher(loadOrg, orgStatus);
+  const {
+    data: organizationData,
```
Comment (jamesrkiger) [inline jsapp/js/account/plans/plan.component.tsx:119]: Suggestion for a general naming convention (in the rest of this PR and the future): When naming component-level variables related to organizations, let's abbreviate to `org`. So `useOrganizationQuery`, for example, would stay as is, but `organizationData` would become `orgData`.

We're going to be adding a lot of org-related code in the near future, so I figure we should figure out these kinds of things now. Let me know what you think.

File: jsapp/js/account/organizations/requireOrgOwner.component.tsx [hunk 3]
```diff
@@ -1,30 +1,34 @@
...
 interface Props {
   children: React.ReactNode;
   redirect?: boolean;
 }
 
 export const RequireOrgOwner = ({children, redirect = true}: Props) => {
-  const [organization, _, orgStatus] = useContext(OrganizationContext);
+  const {
+    data: organizationData,
+    status: organizationStatus,
```
[the remark this answers is not in the document]
Comment (p2edwards) [reply jsapp/js/account/organizations/requireOrgOwner.component.tsx:16]: I think this is a good case for something like `orgQuery.status` and orgQuery.data`

File: jsapp/js/account/organizations/requireOrgOwner.component.tsx [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
+  } = useOrganizationQuery();
   const navigate = useNavigate();
 
   useEffect(() => {
     if (
       redirect &&
-      !orgStatus.pending &&
-      organization &&
-      !organization.is_owner
+      !(organizationStatus === ReactQueryStatuses.pending) &&
```
Comment (p2edwards) [inline jsapp/js/account/organizations/requireOrgOwner.component.tsx:23]: I think we should just use `'pending'` rather than importing QueryStatus here

Comment (jamesrkiger) [reply jsapp/js/account/organizations/requireOrgOwner.component.tsx:23]: @p2edwards That's OK with me. Using QueryStatus does give us a bit more type safety in the event they ever decide to change their status names, but presumably we'd see that in a breaking change note when updating the package.

File: jsapp/js/account/accountSidebar.tsx [hunk 5]
```diff
@@ -45,9 +46,10 @@ function AccountSidebar() {
     setShowPlans(true);
   }, [subscriptionStore.isInitialised]);
 
-  const showAddOnsLink = useMemo(() => {
-    return !subscriptionStore.planResponse.length;
-  }, [subscriptionStore.isInitialised]);
+  const showAddOnsLink = useMemo(
+    () => !subscriptionStore.planResponse.length,
+    [subscriptionStore.isInitialised]
+  );
```
[the remark this answers is not in the document]
Comment (duvld) [reply jsapp/js/account/accountSidebar.tsx:52]: Probably caught by the linter on save, should I remove it?

File: jsapp/js/constants.ts (hunk 1 above)
Comment (duvld) [reply jsapp/js/constants.ts:626]: Didn't know about this :D I'll remove it in favour of `QueryStatus`

File: jsapp/js/account/plans/plan.component.tsx (hunk 2 above)
Comment (duvld) [reply jsapp/js/account/plans/plan.component.tsx:119]: I like doing something like `orgQuery = useOrganizationQuery();` and get the separate data/status/etc. through `orgQuery.data/status/etc.`

File: jsapp/js/query/queries/organizations.query.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import {useQuery} from '@tanstack/react-query';
+import {getOrganization} from 'js/account/stripe.api';
+
+const loadOrganization = async () => {
+  const response = await getOrganization();
+  return response?.results?.[0];
+};
```
[the remark this answers is not in the document]
Comment (duvld) [reply jsapp/js/query/queries/organizations.query.ts:7]: @Akuukis is this what you mean?
```
export const useOrganizationQuery = () => useQuery({
  queryFn: async () => (await getOrganization())?.results?.[0],
  queryKey: ['organization'],
});
```

File: jsapp/js/account/accountSidebar.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Akuukis) [reply jsapp/js/account/accountSidebar.tsx:50]: Yes, it's unrelated to the description. Create a small separate PR instead if you want.

File: jsapp/js/query/queries/organizations.query.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Akuukis) [reply jsapp/js/query/queries/organizations.query.ts:7]: In that case `getOrganization` is still used only in one place. I meant something like this:
```ts
export const useOrganizationQuery = () => useQuery({
  queryFn: async () => {
    const response = await fetchGet<PaginatedResponse<Organization>>(endpoints.ORGANIZATION_URL, {
      errorMessageDisplay: t("Couldn't get data for your organization."),
    });
    return response.results?.[0]
  },
  queryKey: ['organization'],
});
```

[the remark this answers is not in the document]
Comment (duvld) [reply jsapp/js/query/queries/organizations.query.ts:7]: @Akuukis  If I understand correctly we want to remove `getOrganization` from the `stripe.api.ts` file and put that function directly into the query function. Is the justification for this 1. it is only used in 1 place and 2. that it's a single fetch with no other logic?

If that's the case should we be doing the same for the rest of the `stripe.api.ts` file? `postCheckout`, `getProducts`, etc. I think all meet that criteria above. IMO I think it's cleaner leaving it as `getOrganization` and bundling it in a file with other stripe api calls so we know the context for these functions at a glace

[the remark this answers is not in the document]
Comment (duvld) [reply jsapp/js/query/queries/organizations.query.ts:7]: Just had a call with James and Phil and the idea of the change makes more sense to me now :) can disregard the above.

File: jsapp/js/hooks/useApiFetcher.hook.ts [hunk 7]
```diff
@@ -102,6 +111,12 @@ export const useApiFetcher = <Type>(
   return [response, loadFetcher, {...status, setIsInitialLoad}];
 };
 
+/**
+ * *NOTE*: this is deprecated now. Try to use react-query instead.
+ * See `useOrganizationQuery` and its instances for a simple example.
+ *
+ * @deprecated
```
[the remark this answers is not in the document]
Comment (Akuukis) [reply jsapp/js/hooks/useApiFetcher.hook.ts]: Also, see the same above for `useApiFetcher`.

File: jsapp/js/query/queries/organizations.query.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Akuukis) [reply jsapp/js/query/queries/organizations.query.ts:7]: Oh, we should use the basic fetch there, to avoid non-cancellable toast.

**Review by Akuukis — APPROVED** [review]

File: jsapp/js/account/stripe.api.ts [hunk 8]
```diff
@@ -47,11 +48,13 @@ export async function changeSubscription(
...
 }
 
-export async function getOrganization() {
-  return fetchGet<PaginatedResponse<Organization>>(endpoints.ORGANIZATION_URL, {
-    errorMessageDisplay: t("Couldn't get data for your organization."),
-  });
-}
+export const useOrganizationQuery = () => useQuery({
+  queryFn: async () => {
+    const response = await fetchGet<PaginatedResponse<Organization>>(endpoints.ORGANIZATION_URL);
```
Comment (pauloamorimbr) [inline jsapp/js/account/stripe.api.ts:53]: Do we need `PaginatedResponse` here? 🤔

File: jsapp/js/account/plans/plan.component.tsx [hunk 9]
```diff
@@ -115,8 +114,7 @@ export default function Plan(props: PlanProps) {
   >([]);
   const [products, loadProducts, productsStatus] = useContext(ProductsContext);
   useRefreshApiFetcher(loadProducts, productsStatus);
-  const [organization, loadOrg, orgStatus] = useContext(OrganizationContext);
-  useRefreshApiFetcher(loadOrg, orgStatus);
+  const orgQuery = useOrganizationQuery();
```
Comment (pauloamorimbr) [inline jsapp/js/account/plans/plan.component.tsx:117]: Wanna establish that we're gonna always use a QueryResult like this without destructuring?

File: jsapp/js/hooks/useApiFetcher.hook.ts [hunk 10]
```diff
@@ -102,6 +105,9 @@ export const useApiFetcher = <Type>(
   return [response, loadFetcher, {...status, setIsInitialLoad}];
 };
 
+/**
```
Comment (pauloamorimbr) [inline jsapp/js/hooks/useApiFetcher.hook.ts:108]: `withApiFetcher` is used by class components, so maybe we should add more instructions on whether the component should be converted to functional. I don't think any new implementation would use this, tho.

File: jsapp/js/account/stripe.api.ts (hunk 8 above)
Comment (pauloamorimbr) [inline jsapp/js/account/stripe.api.ts:53]: Do we need `PaginatedResponse` here? 🤔

Comment (Akuukis) [reply jsapp/js/account/stripe.api.ts:53]: Yes, it was there before, and implementation does take first element from an array.

File: jsapp/js/account/billingContextProvider.component.tsx [hunk 11]
```diff
@@ -1,26 +1,22 @@
...
   if (!sessionStore.isLoggedIn) {
     return <>{props.children}</>;
   }
-  const [organization, reloadOrg, orgStatus] = useOrganization();
-  const usage = useUsage(organization?.id);
+  const usage = useUsage(orgQuery.data?.id || null);
   const products = useProducts();
   return (
-    <OrganizationContext.Provider value={[organization, reloadOrg, orgStatus]}>
       <UsageContext.Provider value={usage}>
```
Comment (magicznyleszek) [inline jsapp/js/account/billingContextProvider.component.tsx:16]: This should be reindented after wrapper node was removed

**Review by magicznyleszek — CHANGES_REQUESTED** [review]

**Review by pauloamorimbr — APPROVED** [review]
LGTM 👌🏻

File: jsapp/js/account/billingContextProvider.component.tsx (hunk 11 above)
Comment (Akuukis) [reply jsapp/js/account/billingContextProvider.component.tsx]: good catch :ok_hand: 

There is several linter changes in every file, so I got carried away hitting "save without formatting" everywhere. Frankly linter didn't bother about this indent. Anyways that's out of scope until we figure out linter for the project as a whole

**Review by jamesrkiger — APPROVED** [review]

**Pull request merged**